docs/get-started/3d-mesh/index.md

```diff
@@ -0,0 +1,303 @@
+
+```shell
+git clone https://github.com/jina-ai/executor-3d-encoder.git
+```
+
+We also import necessary libraries:
+
+```python
+import pathlib
+from functools import partial
+
+import numpy as np
+import torch
+from docarray import Document, DocumentArray
+from models import MeshDataModel
+
+import finetuner
+from finetuner.tuner.pytorch.losses import TripletLoss
+from finetuner.tuner.pytorch.miner import TripletEasyHardMiner
+```
+
+We define some helper functions for preprocessing and sampling. We convert every `Document` into a point cloud
+tensor using `.load_uri_to_point_cloud_tensor`, we can also pass the number of point you want, here we pass 2048.
+
+Along with loading point clouds, we also assign the `finetuner_label` as a tag. This label will be later used during finetuning.
+
+```python
+def random_sample(pc, num):
+    permutation = np.arange(len(pc))
+    np.random.shuffle(permutation)
+    pc = np.array(pc).astype('float32')
+    pc = pc[permutation[:num]]
+    return pc
+
+
+
+def preprocess(doc: 'Document', num_points: int = 1024, data_aug: bool = True):
+    doc.load_uri_to_point_cloud_tensor(2048)
+    doc.tags['finetuner_label'] = doc.uri.split('/')[2]
+    points = random_sample(doc.tensor, num_points)
+
+    points = points - np.expand_dims(np.mean(points, axis=0), 0)  # center
+    dist = np.max(np.sqrt(np.sum(points ** 2, axis=1)), 0)
+    points = points / dist  # scale
+
+    if data_aug:
+        theta = np.random.uniform(0, np.pi * 2)
+        rotation_matrix = np.array(
+            [[np.cos(theta), -np.sin(theta)], [np.sin(theta), np.cos(theta)]]
+        )
+        points[:, [0, 2]] = points[:, [0, 2]].dot(rotation_matrix)  # random rotation
+        points += np.random.normal(0, 0.02, size=points.shape)  # random jitter
+    return points
+```
+
+In the following code snippet we create MeshData model which encapsulates a `PointConv` model with 512 dimensions, we then load
+training and evaluation data from the binary files we saved before. We create an optimizer and a learning rate scheduler. In this case with use
+an Adam optimizer and MultiStepLR scheduler but you can change those depending on your data and preferences.
+
+```python
+model = MeshDataModel(model_name='pointconv', embed_dim=512) # create pointconv model with 512 dimensions
+
+train = DocumentArray.load_binary('train_data_modelnet.bin') # load train dataset
+eval = DocumentArray.load_binary('test_data_modelnet.bin') # load eval dataset
+
+def configure_optimizer(model):
+    from torch.optim import Adam
+    from torch.optim.lr_scheduler import MultiStepLR
+    # create Adam optimizer with MultistepLR scheduler
+    optimizer = Adam(model.parameters(), lr=5e-4)
+    scheduler = MultiStepLR(optimizer, milestones=[30, 60], gamma=0.5)
+
+    return optimizer, scheduler
+```
+
+We leverage finetuner's mining strategy to improve and finetune the embeddings. By this we mean that the finetuner will use
```

Comment:
Since we are using `TripletEasyHardMiner`, can you explain why it is called triplet and how to get the anchor-positive-negative triplets?

Reply:
I don't want to give a small and insufficient answer so I added a phrase if they want to learn more I redirect them to that section in finetuner docs